gems/active_model_serializers/0.10/_test/test.rbs

```diff
@@ -0,0 +1,29 @@
+module Test
+  class Article < ActiveModelSerializers::Model
+    module GeneratedAttributeMethods
+      attr_accessor title: String
+      attr_accessor body: String
+      attr_accessor tags: Array[Tag]
+
+      def initialize: (title: String, body: String, tags: Array[Tag]) -> void
+    end
+    include GeneratedAttributeMethods
+  end
+
+  class Tag < ActiveModelSerializers::Model
+    module GeneratedAttributeMethods
+      attr_accessor name: String
+
+      def initialize: (name: String) -> void
+    end
+    include GeneratedAttributeMethods
```

Comment:
I thought the following implementation would work, but an error occurred.

```suggestion
    attr_accessor name: String

    def initialize: (name: String) -> void
```

```
test.rb:8:8: [error] Cannot find implementation of method `::Test::Tag#initialize`
│ Diagnostic ID: Ruby::MethodDefinitionMissing
│
└   class Tag < ActiveModelSerializers::Model
          ~~~
```

The current implementation is used because no error occurred when the module was used in reference to the rbs file generated by rbs_rails.

Reply:
Yeah, we can just reuse the `Model#initialize` implementation but with extra constraints with type definition. So, simply put `@dynamic` annotation works.